### PR TITLE
[JBPM-9811] Oracle - Long running query at kie-server startup

### PR DESCRIPTION
**Pull request by gmunozfe**
JIRA: [JBPM-9811](https://issues.redhat.com/browse/JBPM-9811)

### CODE REVIEW

**Review by fjtirado — APPROVED** [review]
Please include hibernate case on description

File: kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java [hunk 1]
```diff
@@ -101,7 +101,11 @@ public String getOffsetLimitSQL(Select select) {
         int limit = select.getLimit();
         StringBuilder out = new StringBuilder();
         if (offset > 0) out.append(" OFFSET ").append(offset).append(" ROWS");
-        if (limit >= 0) out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
+        if (limit >= 0) {
+            out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
+            if (offset==0) out.append(" OFFSET 0 ROWS");
```
Comment (fjtirado) [inline kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java:106]: missing {}

Comment (gmunozfe) [reply kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java]: added

File: kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java [hunk 2]
```diff
@@ -100,8 +100,12 @@ public String getOffsetLimitSQL(Select select) {
...
         int limit = select.getLimit();
         StringBuilder out = new StringBuilder();
-        if (offset > 0) out.append(" OFFSET ").append(offset).append(" ROWS");
-        if (limit >= 0) out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
+        if (offset >= 0) {
+            out.append(" OFFSET ").append(offset).append(" ROWS");
+        }
+        if (limit >= 0) {
+            out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
+        }
```
Comment (afalhambra) [inline kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java:108]: by default, `offset` for OracleDialect is set to `-1`. So this condition is not met at startup and it's only adding `FETCH` to the query.

Maybe we could add a third check here for this particular scenario?

```suggestion
        if (offset == -1 && limit == 0) {
            out.append(" OFFSET 0 ROWS FETCH FIRST 0 ROWS ONLY");
        } else { 
            if (offset >= 0) {
                out.append(" OFFSET ").append(offset).append(" ROWS");
            }
            if (limit >= 0) {
                out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
            }        
        }
```

Comment (fjtirado) [reply kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java:108]: Hmmm, I will use 
```
        if (limit == 0) {
        	return " OFFSET 0 ROWS FETCH FIRST 0 ROWS ONLY";
        } else {
        	StringBuilder out = new StringBuilder();
        	if (offset> 0) out.append(" OFFSET ").append(offset).append(" ROWS");
            if (limit > 0) out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
            return out.toString();
        }
```

**Review by afalhambra — APPROVED** [review]
awesome work!

**Review by MarianMacik — APPROVED** [review]

**Comment by MarianMacik** [comment]
Looks good, merging. Thanks!

**Pull request merged**